syntax = "proto3";

message Product {
  int32 id = 1;
  string name = 2;
  float price = 3;
  string category = 4;
}

message ProductRequest {
  int32 id = 1;
}

message ProductsResponse {
  repeated Product products = 1;
}

service CatalogService {
  rpc GetAllProducts(Empty) returns (ProductsResponse) {}
  rpc GetProduct(ProductRequest) returns (Product) {}
}

message Empty {}